src/test/scala/com/fulcrumgenomics/commons/io/PathUtilTest.scala

```diff
@@ -82,6 +82,21 @@ class PathUtilTest extends UnitSpec {
     PathUtil.replaceExtension(PathUtil.pathTo("/Foo/bar/splat.bam"), ".not_yo_mama") should be (PathUtil.pathTo("/Foo/bar/splat.not_yo_mama"))
   }
 
+  it should "truncate file names longer than 255 characters" in {
+    PathUtil.sanitizeFileName("thisisthemostridiculouslylongfilenameandforsomereasonwestillneedtoh" +
+      "aveasanitizemethodthatensuresthatnoonecreatesafilenamethatisaslongasthisonewhichiskindofsilywhenyou" +
+      "thinkaboutitsincewhywouldyoueverreallyneedafilenamethatisaslongasthisonehonestlyihavenoidea")should be("this" +
+      "isthemostridiculouslylongfilenameandforsomereasonwestillneedtohaveasanitizemethodthatensuresthatnoonecreatesa" +
+      "filenamethatisaslongasthisonewhichiskindofsilywhenyouthinkaboutitsincewhywouldyoueverreallyneedafilenamethat" +
+      "isaslongasthisonehonestlyihavenoi")
+    PathUtil.sanitizeFileName("!!thisisthemostridiculouslylongfilenameandforsomereasonwestillneedtoh" +
+      "aveasanitizemethodthatensuresthatnoonecreatesafilenamethatisaslongasthisonewhichiskindofsilywhenyou" +
+      "thinkaboutitsincewhywouldyoueverreallyneedafilenamethatisaslongasthisonehonestlyihavenoidea",
+      replacement = Some('X')) should be("XXthisisthemostridiculouslylongfilenameandforsomereasonwestillneedtohavea" +
+      "sanitizemethodthatensuresthatnoonecreatesafilenamethatisaslongasthisonewhichiskindofsilywhenyouthinkabout" +
+      "itsincewhywouldyoueverreallyneedafilenamethatisaslongasthisonehonestlyihaven")
```

Comment:
Ooff, I think this test would be clearer what it was testing if you constructed the string with `MaxFileNameSize`.  For example:
```scala
PathUtil.sanitizeFileName("*" + (PathUtil.MaxFileNameSize + 10)) shouldBe "*" + PathUtil.MaxFileNameSize
PathUtil.sanitizeFileName("!" + (PathUtil.MaxFileNameSize + 10), replacement = Some('X')) shouldBe "X" * PathUtil.MaxFileNameSize